src/Interpolations.jl

```diff
@@ -28,11 +28,12 @@ using WoodburyMatrices, Ratios, AxisAlgorithms
 import Base: convert, size, getindex, gradient, promote_rule
 
 abstract InterpolationType
+immutable NoInterp <: InterpolationType end
 abstract GridType
 immutable OnGrid <: GridType end
 immutable OnCell <: GridType end
 
-typealias DimSpec{T} Union(T,Tuple{Vararg{T}})
+typealias DimSpec{T} Union(T,Tuple{Vararg{Union(T,NoInterp)}},NoInterp)
```

Comment:
On the latest release-0.4 branch, it already is. See, e.g., the slew of warnings in https://travis-ci.org/timholy/Images.jl/jobs/80997593. If you update your julia, you'll find a lot here, too.